### PR TITLE
docs: update README with more context

### PR DESCRIPTION
**Pull request by hanno-jonlay**
I don't think this PR is 100% accurate, but just opening an initial version with improved documentation, which we are migrating from one of our internal repositories.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,53 @@
...
+To make use of the action on a Hanno project, you'll want to make the following changes to your source project:
+
+### Adjust your package configuration
+
+This determines which files will be included in the package:
+
+```
+{
+  ...
+  "files": [
```
Comment (hanno-jonlay) [inline README.md:38]: We might perhaps also need to document that you need to configure the:

```
"publishConfig": {
  "registry": "https://npm.pkg.github.com"
},
```

File: README.md [hunk 2]
```diff
@@ -1,13 +1,53 @@
...
+```
+
+### Customise your workflow
+
+In your `.github/workflow.yml`, add the following step:
 
 ```
-- name: Release action
+- name: Release
   uses: @wearehanno/action-semantic-release
```
Comment (hanno-jonlay) [inline README.md:51]: Not sure if we wish to include the `@2` or equivalent version number. Somewhat hard to keep in sync, but I believe it's GitHub best practice to require this.

We could always add this and put a comment next to it to remind the user to look up the most appropriate version?

File: README.md [hunk 3]
```diff
@@ -1,13 +1,53 @@
...
   env:
     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (hanno-jonlay) [inline README.md:53]: `GITHUB_TOKEN` is OK, but I believe `${{ secrets.NPM_TOKEN }}` needs fixing, as per other comments.

Comment (hanno-marcelkalveram) [reply README.md:53]: I've created another PR for this: https://github.com/wearehanno/action-semantic-release/pull/2

File: README.md (hunk 2 above)
Comment (hanno-marcelkalveram) [reply README.md:51]: Ok, let's create a @v2 release as soon as this is fully functional. We'll then update the documentation accordingly when a new release gets pushed. Can't think of a better workflow for this for now.

File: README.md (hunk 1 above)
Comment (hanno-marcelkalveram) [reply README.md:38]: Ok, wanna add this? I previously wasn't sure whether we want to publish to NPM as well, but if we exclusively use GH packages, then yes, we'l have to add this I guess.

**Pull request merged**